### PR TITLE
vli: Skip probing the Dell DA300 device

### PR DESCRIPTION
**Pull request by hughsie**
Although 2109:2820 and 2109:0820, the firmware revision of 3003 is
specific to that hardware.

Fixes https://github.com/fwupd/fwupd/issues/4305

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — DISMISSED** [review]

**Comment by hughsie** [comment]
@superm1 have you tested this, or should we wait for @CragW to test on actual hardware?

**Comment by superm1** [comment]
Wait for actual hardware test. My review is code only

**Comment by superm1** [comment]
Wait for actual hardware test. My review is code only

**Review by CragW — APPROVED** [review]

**Comment by hughsie** [comment]
@CragW it works?! In which case good job everybody.

**Comment by CragW** [comment]
> @CragW it works?! In which case good job everybody.

Yes, that works on Ubuntu and Chrome. Great job!

File: libfwupdplugin/fu-device.c [hunk 1]
```diff
@@ -237,6 +237,8 @@ fu_device_internal_flag_to_string(FuDeviceInternalFlags flag)
 		return "md-set-vendor";
 	if (flag == FU_DEVICE_INTERNAL_FLAG_NO_LID_CLOSED)
 		return "no-lid-closed";
+	if (flag == FU_DEVICE_INTERNAL_FLAG_DISABLED)
+		return "disabled";
```
Comment (hughsie) [inline libfwupdplugin/fu-device.c:241]: @superm1 I wonder if `FU_DEVICE_INTERNAL_FLAG_IGNORE` might be more appropriate here?

Comment (hughsie) [reply libfwupdplugin/fu-device.c:241]: Ya, lets do FU_DEVICE_INTERNAL_FLAG_IGNORE -- it's saying that the device _should be_ ignored, not that it's _in_ a disabled state.

**Comment by hughsie** [comment]
I think FWUPD_ERROR_NOT_SUPPORTED is already a g_debug, no?

**Comment by CragW** [comment]
> I think FWUPD_ERROR_NOT_SUPPORTED is already a g_debug, no?

Maybe it's the engine failed probing a device thus it's a warning here.
Screenshot from 2022-02-22 22-44-25

**Review by CragW — DISMISSED** [review]

**Comment by superm1** [comment]
> > I think FWUPD_ERROR_NOT_SUPPORTED is already a g_debug, no?
> 
> Maybe it's the engine failed probing a device thus it's a warning here.
> Screenshot from 2022-02-22 22-44-25
> 
> 

Do you have the verbose debugging messages turned on or anything? I agree that shouldn't be a warning... It's intentional.

**Comment by CragW** [comment]
> Do you have the verbose debugging messages turned on or anything? I agree that shouldn't be a warning... It's intentional.

It was a normal `get-devices` without verbose output.
```
$ sudo ./dist/fwupdtool.sh get-devices
[sudo] password for crabx: 
Loading…                 [-                                      ]16:09:21:0116 FuEngine             failed to probe device usb:03:00:08: disabled
16:09:22:0250 FuEngine             failed to probe device usb:02:00:01: disabled
Loading…                 [***************************************]
WARNING: This package has not been validated, it may not work properly.
XPS 13 9310
│
```

**Comment by superm1** [comment]
I pushed another patch to this branch to lower the verbosity for that case if you can check it @hughsie .  Otherwise PR LGTM.

**Pull request merged**